### PR TITLE
Dune recap blog post

### PR DESCRIPTION
**Pull request by ryandoowan**

### CODE REVIEW

File: _posts/2023-07-14-trino-fest-2023-dune.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+has 10000s of queries that are executed and saved and re-used each day.
+
+At first, Dune used PostgreSQL where they sharded per blockchain and used a
+vertical scaling. However, they quickly ran into bottleneck issues on storage
+size and IOPS. Thus, they switched to Spark/Databricks for a horizontal scaling
+in order to adopt much more blockchains and to support the vast query volume
+that they had, but this was not interactive enough and was often tied to the
+vendor. In the end, Miguel says in that "Trino was our choice for performance
+reasons and for the good environment and ecosystem and to fully support our
+scheme and our datasets." Using Trino helped their performance issues and made
```
Comment (mosabua) [inline _posts/2023-07-14-trino-fest-2023-dune.md:39]:
```suggestion
scheme and our datasets." Using Trino addressed the performance issues.
```

File: _posts/2023-07-14-trino-fest-2023-dune.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+their queries run faster than previous SQLs.
```
Comment (mosabua) [inline _posts/2023-07-14-trino-fest-2023-dune.md:40]: remove this line

File: _posts/2023-07-14-trino-fest-2023-dune.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+Operating DuneSQL requires more modification and extending Trino to suit the
```
Comment (mosabua) [inline _posts/2023-07-14-trino-fest-2023-dune.md:42]: This whole paragraph is a bit mixed up.

**Review by mosabua — CHANGES_REQUESTED** [review]
MAybe address the comment and then we have a call to chat more about the last big paragraph.

File: _posts/2023-07-14-trino-fest-2023-dune.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+excerpt_separator: <!--more-->
+image: /assets/blog/trino-fest-2023/Dune.png
+show_pagenav: false
+---
+
+The need to make blockchain data easily accessible has risen over the recent
+years due to the rise of cryptocurrencies. Thus, Dune has made this their
+mission: to make crypto data accessible. Dune is a community data platform for
+querying public blockchain data and building beautiful dashboards. They use
+their own query engine called DuneSQL, a fork of TrinoSQL, to query blockchain
```
Comment (colebow) [inline _posts/2023-07-14-trino-fest-2023-dune.md:14]:
```suggestion
their own query engine called DuneSQL, a fork of Trino, to query blockchain
```

File: _posts/2023-07-14-trino-fest-2023-dune.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+data. In Miguel and Jonas' talk, they talk about the challenges in querying
```
Comment (colebow) [inline _posts/2023-07-14-trino-fest-2023-dune.md:15]:
```suggestion
data. In the talk, Miguel and Jonas from Dune talk about the challenges in querying
```

File: _posts/2023-07-14-trino-fest-2023-dune.md [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+
+<!--more-->
+
+{% youtube sCJncarnGdU %}
+
+## Recap
+
+A community data platform is a serverless, open access, community wide
+collaboration portal or application that Dune uses for their query engine.
+However, they have some difficulties with blockchain data such as processing and
```
Comment (colebow) [inline _posts/2023-07-14-trino-fest-2023-dune.md:27]:
```suggestion
However, they have some difficulties with blockchain data, such as processing and
```

File: _posts/2023-07-14-trino-fest-2023-dune.md [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+ingesting raw data, deserializing and decoding function calls and arguments, and
+allowing the community to build abstractions on top. In addition, their engine
+has 10000s of queries that are executed and saved and re-used each day.
```
Comment (colebow) [inline _posts/2023-07-14-trino-fest-2023-dune.md:30]:
```suggestion
has tens of thousands of queries that are executed, saved, and re-used each day.
```

File: _posts/2023-07-14-trino-fest-2023-dune.md [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+
+At first, Dune used PostgreSQL where they sharded per blockchain and used a
```
Comment (colebow) [inline _posts/2023-07-14-trino-fest-2023-dune.md:32]:
```suggestion
At first, Dune used PostgreSQL, where they sharded per blockchain and used a
```

File: _posts/2023-07-14-trino-fest-2023-dune.md [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+vertical scaling. However, they quickly ran into bottleneck issues on storage
+size and IOPS. Thus, they switched to Spark/Databricks for a horizontal scaling
```
Comment (colebow) [inline _posts/2023-07-14-trino-fest-2023-dune.md:34]: IOPS probably needs to be spelled out, maybe in parentheses. Not an acronym you encounter super frequently.

File: _posts/2023-07-14-trino-fest-2023-dune.md [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+in order to adopt much more blockchains and to support the vast query volume
```
Comment (colebow) [inline _posts/2023-07-14-trino-fest-2023-dune.md:35]:
```suggestion
in order to adopt more blockchains and to support the vast query volume
```

File: _posts/2023-07-14-trino-fest-2023-dune.md [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+that they had, but this was not interactive enough and was often tied to the
```
Comment (colebow) [inline _posts/2023-07-14-trino-fest-2023-dune.md:36]:
```suggestion
that they had. Unfortunately, this was not interactive enough and was often tied to the
```
Let's break it up into two sentences.

File: _posts/2023-07-14-trino-fest-2023-dune.md [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+vendor. In the end, Miguel says in that "Trino was our choice for performance
```
Comment (colebow) [inline _posts/2023-07-14-trino-fest-2023-dune.md:37]:
```suggestion
vendor. In the end, Miguel says in that, "Trino was our choice for performance
```

File: _posts/2023-07-14-trino-fest-2023-dune.md (hunk 2 above)
Comment (colebow) [inline _posts/2023-07-14-trino-fest-2023-dune.md:40]:
```suggestion
their queries run faster than previous SQL engines.
```

File: _posts/2023-07-14-trino-fest-2023-dune.md (hunk 3 above)
Comment (colebow) [inline _posts/2023-07-14-trino-fest-2023-dune.md:42]:
```suggestion
Operating DuneSQL requires modifications and extensions of Trino to suit the
```

File: _posts/2023-07-14-trino-fest-2023-dune.md [hunk 13]
```diff
@@ -0,0 +1,59 @@
...
+reasons and for the good environment and ecosystem and to fully support our
+scheme and our datasets." Using Trino helped their performance issues and made
+their queries run faster than previous SQLs.
+
+Operating DuneSQL requires more modification and extending Trino to suit the
+needs of the users and platform as a whole. DuneSQL is able to handle 10000s of
```
Comment (colebow) [inline _posts/2023-07-14-trino-fest-2023-dune.md:43]: tens of thousands again

File: _posts/2023-07-14-trino-fest-2023-dune.md [hunk 14]
```diff
@@ -0,0 +1,59 @@
...
+queries per day and be able to provide the performance for different tiers. In
+addition, they use query execution services and Trino-Gateway to handle the
```
Comment (colebow) [inline _posts/2023-07-14-trino-fest-2023-dune.md:45]: Embed a hyperlink to gateway: https://github.com/lyft/presto-gateway

File: _posts/2023-07-14-trino-fest-2023-dune.md [hunk 15]
```diff
@@ -0,0 +1,59 @@
...
+scheduling and load balancing. With clusters, they have a fixed size in order
+for it to have a predictable capacity and performance, and to reduce the
+blast-radius so it does not affect other clusters. Even with all this, they have
+much work to do as they plan to optimize the billions of S3 requests they
+receive, improve data layout, and implementing sandboxed user defined functions.
```
Comment (colebow) [inline _posts/2023-07-14-trino-fest-2023-dune.md:50]:
```suggestion
receive, improve data layout, and implement sandboxed user defined functions.
```
Keep tense consistent.

**Review by colebow — CHANGES_REQUESTED** [review]

File: _posts/2023-07-14-trino-fest-2023-dune.md [hunk 16]
```diff
@@ -0,0 +1,63 @@
...
+title: "DuneSQL - A query engine for blockchain data"
+author: "Miguel Filipe, Jonas Irgens Kylling, Ryan Duan"
+excerpt_separator: <!--more-->
+image: /assets/blog/trino-fest-2023/Dune.png
+show_pagenav: false
+---
+
+The need to make blockchain data easily accessible has risen over the recent
+years due to the rise of cryptocurrencies, NFTs, and other uses of blockchains.
+Dune has made this their mission to make crypto data more accessible. Dune is a
```
Comment (colebow) [inline _posts/2023-07-14-trino-fest-2023-dune.md:12]:
```suggestion
Dune has made it their mission to make crypto data more accessible. Dune is a
```

File: _posts/2023-07-14-trino-fest-2023-dune.md [hunk 17]
```diff
@@ -0,0 +1,63 @@
...
+about the challenges of querying blockchain data, their transition to Trino, and
+how DuneSQL is operated. Give the talk a listen or keep reading for a recap.
+
+<!--more-->
+
+{% youtube sCJncarnGdU %}
+
+## Recap
+
+The Dune community data platform is a serverless, open access, community wide
```
Comment (colebow) [inline _posts/2023-07-14-trino-fest-2023-dune.md:25]:
```suggestion
The Dune community data platform is a serverless, open access, community-wide
```

File: _posts/2023-07-14-trino-fest-2023-dune.md [hunk 18]
```diff
@@ -0,0 +1,63 @@
...
+
+In addition, operating DuneSQL requires modifications and extensions of Trino to
+suit the needs of the users and platform as a whole. DuneSQL needs to manage the
+whole fleet and the capacity they have, because they use over 4000 CPUs per
+hour, do more than 100 billion S3 requests per month, and operate over 10
+clusters. To handle the scheduling and load balancing of these massive
+operations, DuneSQL use query execution services and
+[Trino-Gateway](https://github.com/lyft/presto-gateway). To manage clusters,
+they make it have a fixed size in order for it to have a predictable capacity
+and performance, and to reduce the blast-radius so it does not affect other
```
Comment (colebow) [inline _posts/2023-07-14-trino-fest-2023-dune.md:51]:
```suggestion
and performance and to reduce the blast-radius so it does not affect other
```

**Review by colebow — APPROVED** [review]
A few tiny nits

File: _posts/2023-07-14-trino-fest-2023-dune.md [hunk 19]
```diff
@@ -0,0 +1,63 @@
...
+layout: post
+title: "DuneSQL - A query engine for blockchain data"
+author: "Miguel Filipe, Jonas Irgens Kylling, Ryan Duan"
+excerpt_separator: <!--more-->
+image: /assets/blog/trino-fest-2023/Dune.png
+show_pagenav: false
+---
+
+The need to make blockchain data easily accessible has risen over the recent
+years due to the rise of cryptocurrencies, NFTs, and other uses of blockchains.
```
Comment (mosabua) [inline _posts/2023-07-14-trino-fest-2023-dune.md:11]:
```suggestion
years due to the popularity of cryptocurrencies, NFTs, and other uses of blockchains.
```

File: _posts/2023-07-14-trino-fest-2023-dune.md [hunk 20]
```diff
@@ -0,0 +1,63 @@
...
+Dune has made it their mission to make crypto data more accessible. Dune is a
```
Comment (mosabua) [inline _posts/2023-07-14-trino-fest-2023-dune.md:12]:
```suggestion
Dune has made it their mission to make blockchain data more accessible. Dune is a
```

File: _posts/2023-07-14-trino-fest-2023-dune.md [hunk 21]
```diff
@@ -0,0 +1,63 @@
...
+community data platform for querying public blockchain data and building
+beautiful dashboards. They use their own query engine called DuneSQL, a fork of
```
Comment (mosabua) [inline _posts/2023-07-14-trino-fest-2023-dune.md:14]:
```suggestion
beautiful dashboards. They use their own query engine called DuneSQL, built as extension of 
```

File: _posts/2023-07-14-trino-fest-2023-dune.md [hunk 22]
```diff
@@ -0,0 +1,63 @@
...
+Trino, to query blockchain data. In the talk, Miguel and Jonas from Dune talk
```
Comment (mosabua) [inline _posts/2023-07-14-trino-fest-2023-dune.md:15]:
```suggestion
Trino, to query blockchain data. In the session, Miguel and Jonas from Dune talk
```

File: _posts/2023-07-14-trino-fest-2023-dune.md [hunk 23]
```diff
@@ -0,0 +1,63 @@
...
+about the challenges of querying blockchain data, their transition to Trino, and
+how DuneSQL is operated. Give the talk a listen or keep reading for a recap.
```
Comment (mosabua) [inline _posts/2023-07-14-trino-fest-2023-dune.md:17]:
```suggestion
how DuneSQL is operated. Watch the recording of the session or keep reading for a recap.
```

File: _posts/2023-07-14-trino-fest-2023-dune.md [hunk 24]
```diff
@@ -0,0 +1,63 @@
...
+
+{% youtube sCJncarnGdU %}
+
+## Recap
+
+The Dune community data platform is a serverless, open access, community-wide
+collaboration portal. Dune experienced some difficulties with blockchain data,
+such as processing and ingesting raw data, deserializing and decoding function
+calls and arguments, and allowing the community to build abstractions. In
+addition, their engine, DuneSQL, is Trino with custom extensions that they
```
Comment (mosabua) [inline _posts/2023-07-14-trino-fest-2023-dune.md:29]: Remove "In addition"

File: _posts/2023-07-14-trino-fest-2023-dune.md [hunk 25]
```diff
@@ -0,0 +1,63 @@
...
+created. It runs tens of thousands of queries that are executed, saved, and
+re-used each day.
+
+At first, Dune used PostgreSQL, where they sharded per blockchain and used a
```
Comment (mosabua) [inline _posts/2023-07-14-trino-fest-2023-dune.md:33]:
```suggestion
At first, Dune used PostgreSQL, where they sharded per blockchain and used
```

File: _posts/2023-07-14-trino-fest-2023-dune.md [hunk 26]
```diff
@@ -0,0 +1,63 @@
...
+vertical scaling. However, they quickly ran into bottleneck issues on storage
+size and IOPS (I/O operations per second). Thus, they switched to Apache Spark
+with Databricks to allow horizontal scaling and support more blockchains
+processing and to support the vast query volume that they had. Unfortunately,
+the result was not performant and not interactive enough. In the end, Miguel
+says that, "Trino was our choice for performance reasons, for the good
+environment and ecosystem, and to fully support our scheme and our datasets."
+Using Trino addressed the performance issues.
+
+In addition, operating DuneSQL requires modifications and extensions of Trino to
```
Comment (mosabua) [inline _posts/2023-07-14-trino-fest-2023-dune.md:43]:
```suggestion
Operating DuneSQL requires modifications and extensions of Trino to
```

File: _posts/2023-07-14-trino-fest-2023-dune.md [hunk 27]
```diff
@@ -0,0 +1,63 @@
...
+suit the needs of the users and platform as a whole. DuneSQL needs to manage the
+whole fleet and the capacity they have, because they use over 4000 CPUs per
+hour, do more than 100 billion S3 requests per month, and operate over 10
+clusters. To handle the scheduling and load balancing of these massive
+operations, DuneSQL use query execution services and
```
Comment (mosabua) [inline _posts/2023-07-14-trino-fest-2023-dune.md:48]:
```suggestion
operations, DuneSQL uses query execution services and the
```

File: _posts/2023-07-14-trino-fest-2023-dune.md [hunk 28]
```diff
@@ -0,0 +1,63 @@
...
+[Trino-Gateway](https://github.com/lyft/presto-gateway). To manage clusters,
```
Comment (mosabua) [inline _posts/2023-07-14-trino-fest-2023-dune.md:49]:
```suggestion
[gateway](https://github.com/lyft/presto-gateway).
```

File: _posts/2023-07-14-trino-fest-2023-dune.md [hunk 29]
```diff
@@ -0,0 +1,63 @@
...
+they make it have a fixed size in order for it to have a predictable capacity
```
Comment (mosabua) [inline _posts/2023-07-14-trino-fest-2023-dune.md:50]:
```suggestion
Clusters have a fixed size to have a predictable capacity
```

File: _posts/2023-07-14-trino-fest-2023-dune.md [hunk 30]
```diff
@@ -0,0 +1,63 @@
...
+and performance and to reduce the blast-radius so it does not affect other
```
Comment (mosabua) [inline _posts/2023-07-14-trino-fest-2023-dune.md:51]:
```suggestion
and performance. The gateway exposes the clusters to reduce the blast-radius so failures do not affect other
```

**Review by mosabua — CHANGES_REQUESTED** [review]

**Review by mosabua — APPROVED** [review]

**Pull request merged**